Minor cleanups in planner

martint opened the pull request.

findepi commented on presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java:
```diff
@@ -670,7 +669,7 @@ private RelationPlan planJoinUnnest(RelationPlan leftPlan, Join joinNode, Unnest
         ProjectNode projectNode = (ProjectNode) planBuilder.getRoot();
 
         ImmutableMap.Builder<Symbol, List<Symbol>> unnestSymbols = ImmutableMap.builder();
-        UnmodifiableIterator<Symbol> unnestedSymbolsIterator = unnestedSymbols.iterator();
```
❤️

findepi commented on presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java:
```diff
@@ -831,6 +795,29 @@ else if (type instanceof MapType) {
         return new RelationPlan(unnestNode, scope, unnestedSymbols);
     }
 
+    private void computeUnnestMappings(ImmutableMap.Builder<Symbol, List<Symbol>> mappings, Iterator<Symbol> outputs, Symbol input, Type type)
```
- I don't like Iterator being an argument here.  I now it's preexisting, but extracting method makes it more apparent. I am concerned some day an off by one problem will creep here.

- you can make the method return `Map<..>` (or `Entry`) instead of passing `ImmutableMap.Builder`.

martint replied:
I know. The problem is that the iterator and the map builder are "stateful" across multiple invocations of computeUnnestMappings. Let me see if I can fix that some way.

findepi replied:
ideally we would record the during analysis, instead of matching (semi-)positionally unnest symbols with output descriptor.

martint replied:
Yup, that’s a bigger cleanup I’ve been working on, but didn’t want to complicate this one too much.

martint commented:
> LGTM except last commit (Remove unnecessary translation) which i didn't review.

I replaced the last commit. It's actually needed to be able to swap function references to their resolved form. I added another commit with a comment and a test, instead.

martint commented:
Superseded by https://github.com/prestosql/presto/pull/3496
